src/scancode/resource.py

```diff
@@ -120,6 +120,39 @@ class ResourceNotInCache(Exception):
 class UnknownResource(Exception):
     pass
 
+def depth_walk(root_location, max_depth, skip_ignored=lambda x: False, error_handler=lambda:None):
+    """ Generator that walks the `root_location` directory recursively upto `max_depth` of 
+    nested subdirectories and yields a (top, dirs, files) tuple at each step. 
+
+    Arguments: 
+       - root_location: Absolute path for the directory to be walked
+       - max_depth: Positive integer for fixed depth limit. negative for no limit
+       - skip_ignored: Function that returns boolean indicating whether to ignore searching files
+                      in the current location. No ignoring by default
+       - error_handler: Error handler callback. No action taken by default.
+    """
+
+    # Find root directory depth using path separator's count
+    root_dir_depth = root_location.count(os.path.sep)
+
+    # Default behaviour. current_depth is always less than max_depth so that 
+    # os_walk keeps running. When max_depth is limited, the current_depth is
+    # changed in each iteration and compared with max_depth
+    current_depth = -2
```

Comment:
-2 does not feel right...see below another approach. where the default would be 0.